app/work_template_executor.go

```diff
@@ -0,0 +1,273 @@
+	cBoard *model.WorkTemplateBoard,
+	linkToChannelID string,
+) (string, error) {
+	boardService := e.app.Srv().services[product.BoardsKey].(product.BoardsService)
+	templates, err := boardService.GetTemplates("0", c.Session().UserId)
+	if err != nil {
+		return "", fmt.Errorf("error while getting templates: %w", err)
+	}
+
+	var template *fb_model.Board = nil
+	for _, t := range templates {
+		v, ok := t.Properties["trackingTemplateId"]
+		if ok && v == cBoard.Template {
+			template = t
+			break
+		}
+	}
+	if template == nil {
+		return "", fmt.Errorf("template not found")
+	}
+
+	title := cBoard.Name
+	if wtcr.Name != "" {
+		title = fmt.Sprintf("%s: %s", wtcr.Name, cBoard.Name)
+	}
+
+	// Duplicate board From template
+	boardsAndBlocks, _, err := boardService.DuplicateBoard(template.ID, c.Session().UserId, wtcr.TeamID, false)
+	if err != nil {
+		return "", fmt.Errorf("failed to create new board from template: %w", err)
+	}
+	if len(boardsAndBlocks.Boards) != 1 {
+		return "", fmt.Errorf("only one board was expected, found %d", len(boardsAndBlocks.Boards))
+	}
+
+	// Apply patch for the title and linked channel
+	_, err = boardService.PatchBoard(&fb_model.BoardPatch{
+		Title:     &title,
+		ChannelID: &linkToChannelID,
+	}, boardsAndBlocks.Boards[0].ID, c.Session().UserId)
+	if err != nil {
+		return "", fmt.Errorf("failed to patch board: %w", err)
+	}
+
+	return boardsAndBlocks.Boards[0].ID, nil
+}
+
+func (e *appWorkTemplateExecutor) InstallPlugin(
+	c *request.Context,
+	wtcr *worktemplates.ExecutionRequest,
+	cIntegration *model.WorkTemplateIntegration,
+	sendToChannelID string,
+) error {
+	// check if this plugin is already installed
+	pluginID := cIntegration.ID
+	_, appErr := e.app.GetPluginStatus(pluginID)
+	if appErr != nil {
+		if appErr.Id == "app.plugin.not_installed.app_error" {
+			// we install them in the background as we don't want user to wait for this
+			manifest, installAppErr := e.app.Channels().InstallMarketplacePlugin(&model.InstallMarketplacePluginRequest{
+				Id:      pluginID,
+				Version: "",
+			})
+			if installAppErr != nil {
+				return fmt.Errorf("unable to install plugin: %w", installAppErr)
+			}
+			if sendToChannelID != "" {
+				e.app.SendEphemeralPost(c, c.Session().UserId, &model.Post{
+					ChannelId: sendToChannelID,
+					Message:   fmt.Sprintf("plugin %s has been installed", manifest.Name),
+					CreateAt:  model.GetMillis(),
+				})
+			}
+		} else {
+			return fmt.Errorf("unable to get plugin status: %w", appErr)
+		}
+	}
+
+	// get plugin state
+	if err := e.app.EnablePlugin(pluginID); err != nil {
+		return fmt.Errorf("unable to enable plugin: %w", err)
+	}
+	return nil
+}
+
+type playbookCreateResponse struct {
+	ID string `json:"id"`
+}
+
+type playbookRunCreateResponse struct {
+	ID        string `json:"id"`
+	ChannelID string `json:"channel_id"`
+}
+
+var allNonSpaceNonWordRegex = regexp.MustCompile(`[^\w\s]`)
+
+func cleanChannelName(channelName string) string {
```

Comment:
Actually during my tests I realized that playbook were doing sanitation on the input for the channel, and the code comes directly from them!

Reply:
yeah I'll add a comment.
It would actually be nice to add this to the plugin API and give plugins the ability to reuse this, but I don't wanna my PR to cover even more changes 😅